src/components/StepWrapper/Sidebar.tsx

```diff
@@ -1,202 +1,165 @@
 import getWindowWidth from 'lib/getWindowWidth'
-import {connect} from 'react-redux'
-import {ObjectOf} from 'interfaces'
 import ChapterCursor from './ChapterCursor'
-
 import {
-  ISidebarTranslations,
-  ISidebarTrack,
-  IDawaSidebarTrack,
-} from '../../types/sidebar'
+  TChapterPageTrackFragment,
+  TChapterPageTranslationsFragment,
+  TQuizzesPageTranslationsFragment,
+} from '../../graphqlTypes'
+import {useAllChaptersStates, ChapterStates} from 'hooks/useChapterState'
 
 const GATSBY_IS_DAWA_APP = process.env.GATSBY_IS_DAWA_APP
 
 interface IProps {
   currentCourseSlug: string
-  track: ISidebarTrack | IDawaSidebarTrack
+  track: TChapterPageTrackFragment
+  t: TChapterPageTranslationsFragment | TQuizzesPageTranslationsFragment
   isOpen: boolean
-  toggleDrawer: () => void
-  t: ISidebarTranslations
+  toggleSidebar: () => void
 }
 
-type TQuizState = ObjectOf<
-  ObjectOf<
-    ObjectOf<{
-      passed: boolean
-    }>
-  >
->
-interface IConnectReturn {
-  chaptersState: ObjectOf<boolean>
-  quizsState: TQuizState
-}
+const difficultyStrings: ['difficulty1', 'difficulty2'] = [
+  'difficulty1',
+  'difficulty2',
+]
 
-class Sidebar extends Component<IProps & IConnectReturn> {
-  private closeDrawerOnMobile = (): void => {
-    const {toggleDrawer} = this.props
-    // TODO: don't work, need to use redux for this, right now we don't have any animation
+const Sidebar: React.FC<IProps> = ({
+  currentCourseSlug,
+  track,
+  isOpen,
+  toggleSidebar,
+  t,
+}) => {
+  const closeDrawerOnMobile = (): void => {
     if (getWindowWidth() < 800) {
-      toggleDrawer()
+      toggleSidebar()
     }
   }
 
-  public render(): JSX.Element {
-    const {
-      chaptersState,
-      currentCourseSlug,
-      track,
-      isOpen,
-      quizsState,
-      t,
-    } = this.props
-
-    const {courses} = track
-    return (
-      <aside
-        className={cx(
-          'z-999 fixed w320px min-vh-100 bg-custom-dark-blue flex-no-shrink transition-transform overflow-auto',
-          {
-            'left-0': t.locale !== 'ar',
-            'right-0': t.locale === 'ar',
-            'translate-x-0': isOpen,
-            'translate-x-100': !isOpen && t.locale === 'ar',
-            'translate-x--100': !isOpen && t.locale !== 'ar',
-          },
-        )}
-        style={{height: '100vh'}}
-      >
-        <nav>
-          <header>
-            <Link
-              className="white no-underline"
-              to={
-                GATSBY_IS_DAWA_APP
-                  ? `${t.localePath}`
-                  : `${t.localePath}${track.slug}/`
-              }
-            >
-              <div className="pv2 flex items-center">
-                <img alt="" className="w2-5 rtl-rotate-180" src={leftSvg} />
-                <div className="f4">
-                  {track.translations.edges[0].node.title}
-                </div>
-              </div>
-            </Link>
-          </header>
+  const {courses} = track
```

Comment:
We have to make sure track have a default value to avoid breaking the component, also you can use the memo hook for this to further optimize component rendering